src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/ILGeneratorImpl.cs

```diff
@@ -210,7 +220,32 @@ public override void Emit(OpCode opcode, string str)
             _il.Token(tempVal);
         }
 
-        public override void Emit(OpCode opcode, ConstructorInfo con) => throw new NotImplementedException();
+        public override void Emit(OpCode opcode, ConstructorInfo con)
+        {
+            ArgumentNullException.ThrowIfNull(con);
+
+            if (!(opcode.Equals(OpCodes.Call) || opcode.Equals(OpCodes.Callvirt) || opcode.Equals(OpCodes.Newobj)))
```

Comment:
Runtime implementation Asserts this instead of throwing:
https://github.com/dotnet/runtime/blob/5f0fe49eb2920d3a9ed59e7459c1b2bea503fa90/src/coreclr/System.Private.CoreLib/src/System/Reflection/Emit/RuntimeILGenerator.cs#L680-L694
I think this should throw instead, same as in the `EmitCall(OpCode opcode, MethodInfo methodInfo, Type[]? optionalParameterTypes)`:
https://github.com/dotnet/runtime/blob/5f0fe49eb2920d3a9ed59e7459c1b2bea503fa90/src/coreclr/System.Private.CoreLib/src/System/Reflection/Emit/RuntimeILGenerator.cs#L603-L604